src/xtgeo/grid3d/_egrid.py

```diff
@@ -691,19 +704,21 @@ def map_axis_units(self) -> Units:
         return self.egrid_head.mapunits
 
     @map_axis_units.setter
-    def map_axis_units(self, value):
+    def map_axis_units(self, value: Units) -> None:
         self.egrid_head.mapunits = value
 
     @property
     def grid_units(self) -> Units:
+        assert self.egrid_head.gridunit is not None
         return self.egrid_head.gridunit.unit
 
     @grid_units.setter
-    def grid_units(self, value):
+    def grid_units(self, value: Units) -> None:
+        assert self.egrid_head.gridunit is not None
         self.egrid_head.gridunit.unit = value
 
     @classmethod
-    def from_xtgeo_grid(cls, xtgeo_grid):
+    def from_xtgeo_grid(cls, xtgeo_grid: GridProperty) -> EclGrid:
```

Comment:
Same here, should return an `EGrid` since it's a classmethod. mypy might let it slide because this class derives from `EclGrid` though.